specifyweb/frontend/js_src/lib/components/Molecules/DateElement.tsx

```diff
@@ -19,14 +20,71 @@ export function DateElement({
   readonly fallback?: React.ReactNode;
   readonly flipDates?: boolean;
 }): JSX.Element {
+  const [relativeDate, setRelativeDate] = React.useState<string | undefined>(
+    date ? getRelativeDate(new Date(date)) : undefined
+  );
+
+  React.useEffect(() => {
+    let intervalValue = 0;
+    let interval: NodeJS.Timeout | undefined;
+
+    if (date) {
+      interval = setInterval(() => {
+        const dateObject = new Date(date);
+        const now = new Date();
+        const timeDifference = now.getTime() - dateObject.getTime();
+
+        if (timeDifference < 60 * MILLISECONDS)
+          intervalValue = 5 * MILLISECONDS;
+        else if (timeDifference < 6 * MINUTE) intervalValue = MINUTE;
+        else intervalValue = HOUR;
+
+        setRelativeDate(getRelativeDate(new Date(date)));
+      }, intervalValue);
```

Comment:
wait, setInterval wouldn't work correctly here at all, right?
even though you modify the value of intervalValue, this is not react - setInterval will keep on using the value it was called with the first value - thus you would have to cancel the interval and set a new one - at which point it would be simpler to just use setTimeout